pandas/core/arrays/datetimes.py

```diff
@@ -510,7 +510,11 @@ def _check_compatible_with(self, other, setitem: bool = False):
     # Descriptive Properties
 
     def _box_func(self, x) -> Timestamp | NaTType:
-        return Timestamp(x, freq=self.freq, tz=self.tz)
+        with warnings.catch_warnings():
+            # filter out warnings about Timestamp.freq
+            warnings.filterwarnings("ignore", category=FutureWarning)
+
+            return Timestamp(x, freq=self.freq, tz=self.tz)
```

Comment:
if we could change this without a deprecation cycle itd be convenient, but that would change what you get from `dti[0]`

Reply:
Can you use the internal `_set_freq` after construction instead of the catch_warnings? 
Iteratively calling catch_warnings many times will give a lot of overhead (although we don't use this ourselves in something like `__iter__` where we need to box many objects, but the user could still do something similar themselves with getitem?)